mono/mini/aot-compiler.c

```diff
@@ -7465,6 +7465,9 @@ mono_aot_parse_options (const char *aot_options, MonoAotOptions *opts)
 			opts->mtriple = g_strdup (arg + strlen ("mtriple="));
 		} else if (str_begins_with (arg, "llvm-path=")) {
 			opts->llvm_path = clean_path (g_strdup (arg + strlen ("llvm-path=")));
+		} else if (!strcmp (arg, "try_llvm")) {
+			// If we can load LLVM, use it
```

Comment:
This is true, though if we exit this function without opts->llvm set, that will only push complexity up to whomever has to check the "pending_llvm" and try to load it and try to set it. And we check `->llvm` alone in too many places to just go with another variable. 

I could move the loading before this function, which is what I had considered before. I viewed it as a bit much though, because that will load LLVM in the cases when someone is using the JIT without llvm. 

We could pre-search the string and load if we see any mention of llvm. That seemed hackish to me, but it's perfectly viable as long as it's kept up-to-date.

Reply:
I think I want a warning when this fails. Maybe. It will inevitably be missed when it matters and noisy when it doesn't matter however, and sometimes noticed when it matters.